src/fs/metadata_log_operations/create_file.hh

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+#pragma once
+
+#include "fs/metadata_log.hh"
+#include "fs/path.hh"
+#include "seastar/core/future.hh"
+
+namespace seastar::fs {
+
+class create_file_operation {
+    metadata_log& _metadata_log;
+    bool _is_directory;
+    std::string _entry_name;
+    file_permissions _perms;
+    inode_t _dir_inode;
+    inode_info::directory* _dir_info;
+
+    create_file_operation(metadata_log& metadata_log) : _metadata_log(metadata_log) {}
```

Comment:
This variable needs to initialize `_is_directory` to something, otherwise debug mode warns that sometimes a non-initialized value is used here. You can also put
```cpp
bool _is_directory = true;
```
in the above declaration and be done with all esoteric cases.